src/main/java/pico/typecheck/PICOVisitor.java

```diff
@@ -375,30 +375,6 @@ private void checkNewInstanceCreation(Tree node) {
         }
     }
 
-    @Override
-    public Void visitMethodInvocation(MethodInvocationTree node, Void p) {
-        super.visitMethodInvocation(node, p);
-        ParameterizedExecutableType mfuPair =
-                atypeFactory.methodFromUse(node);
-        AnnotatedExecutableType invokedMethod = mfuPair.executableType;
-        ExecutableElement invokedMethodElement = invokedMethod.getElement();
-        // Only check invocability if it's super call, as non-super call is already checked
-        // by super implementation(of course in both cases, invocability is not checked when
-        // invoking static methods)
-        if (!ElementUtils.isStatic(invokedMethodElement) && TreeUtils.isSuperCall(node)) {
```

Comment:
We recently renamed `TreeUtils.isSuperCall` to `isSuperConstructorCall` to make it more obvious that this method is only for constructor calls.

But is this code maybe related to `super.foo(...)` invocations? Are there any special rules to consider?

Reply:
I think this code is not related to `super.foo(...)`, `TreeUtils.isSuperCall(node)` returns false when `node` is `super.foo()`.

Furthermore, let's see `TreeUtils.isSuperCall(node)`:
```java
    public static boolean isSuperCall(MethodInvocationTree tree) {
        return isNamedMethodCall("super", tree);
    }
    private static boolean isNamedMethodCall(String name, MethodInvocationTree tree) {
        return getMethodName(tree.getMethodSelect()).equals(name);
    }
```
When `super.foo()` is invoked, `getMethodName(tree.getMethodSelect())` equals to `foo`.

Reply:
Sorry, I wasn't clear. I meant that maybe some additional logic for `super.foo()` was intended, but incorrectly a check for super constructor invocations was implemented.
Is there anything additionally that needs checking for PICO in that context?

Reply:
I think there is no any other additional checkings needed. `BaseTypeVisitor.checkMethodInvocability()` is enough.